# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.cpp
# Commit: ced517f952ce
# Full Hash: ced517f952ce8591d012ba92711859ef89a900f7
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2021-04-15 21:46:43
# Description:
#   Bug 1704561 - Ensure that Module return value is an object before binding it to evaluationPromise; r=evilpie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D111849
# ==============================================================================

diff -r b099dbba4587 -r ced517f952ce dom/script/ScriptLoader.cpp
--- a/dom/script/ScriptLoader.cpp	Thu Apr 15 12:11:02 2021 +0000
+++ b/dom/script/ScriptLoader.cpp	Thu Apr 15 12:28:48 2021 +0000
@@ -3207,7 +3207,7 @@
       } else {
         // Path for when Top Level Await is enabled
         JS::Rooted<JSObject*> aEvaluationPromise(cx);
-        if (NS_SUCCEEDED(rv)) {
+        if (rval.isObject()) {
           // If the user cancels the evaluation on an infinite loop, we need
           // to skip this step. In that case, ModuleEvaluate will not return a
           // promise, rval will be undefined. We should treat it as a failed